patches/chromium/fix_svg_filter_data_invalidation_for_empty_containers.patch

```diff
@@ -0,0 +1,118 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Fredrik=20S=C3=B6derqvist?= <fs@opera.com>
+Date: Tue, 1 Dec 2020 17:15:59 +0000
+Subject: Fix SVG filter data invalidation for empty containers
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Empty containers will fail to create FilterData because they have an
+empty bounding box. When children were added to the container, making
+the bounding box non-empty we would fail to mark the paint properties as
+needing update because of the check for existing FilterData in
+SVGElementResourceClient::InvalidateFilterData().
+
+Instead always dispose of the filter data and mark paint properties for
+update, and use the |filter_data_dirty_| flag as an early-out check.
+
+Since we can now call InvalidateFilterData() directly in
+SVGElementResourceClient::ResourceElementChanged(), fold
+ClearFilterData() into InvalidateFilterData().
+
+Bug: 1154050
+Change-Id: I01c8ebac4f62532e2c17aabc3d998d7601dbb71b
+Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2566992
+Reviewed-by: Xianzhu Wang <wangxianzhu@chromium.org>
+Commit-Queue: Fredrik Söderquist <fs@opera.com>
+Cr-Commit-Position: refs/heads/master@{#832391}
+
+diff --git a/third_party/blink/renderer/core/layout/svg/svg_resources.cc b/third_party/blink/renderer/core/layout/svg/svg_resources.cc
+index f37cb5c3052d72ab15e769b29f8456e7ced9b5af..3ccc73b6ff4d5a9cb0c49ec8ea3aae80fbd8b9ca 100644
+--- a/third_party/blink/renderer/core/layout/svg/svg_resources.cc
++++ b/third_party/blink/renderer/core/layout/svg/svg_resources.cc
+@@ -728,13 +728,16 @@ void SVGElementResourceClient::ResourceContentChanged(
+ }
+ 
+ void SVGElementResourceClient::ResourceElementChanged() {
+-  if (LayoutObject* layout_object = element_->GetLayoutObject()) {
+-    ClearFilterData();
+-    SVGResourcesCache::ResourceReferenceChanged(*layout_object);
+-    // TODO(fs): If the resource element (for a filter) doesn't actually change
+-    // we don't need to perform the associated invalidations.
+-    layout_object->SetNeedsPaintPropertyUpdate();
+-    MarkFilterDataDirty();
++  LayoutObject* layout_object = element_->GetLayoutObject();
++  if (!layout_object)
++    return;
++  // TODO(fs): If the resource element (for a filter) doesn't actually change
++  // we don't need to perform the associated invalidations.
++  InvalidateFilterData();
++  if (layout_object->Parent()) {
```

Comment:
This codepath was only added as a refactor from https://chromium-review.googlesource.com/c/chromium/src/+/2489650

It is not available in this branch, the change should be

```
LayoutObject* layout_object = element_->GetLayoutObject();
if (!layout_object)
    return;
// TODO(fs): If the resource element (for a filter) doesn't actually change
// we don't need to perform the associated invalidations.
InvalidateFilterData();
SVGResourcesCache::ResourceReferenceChanged(*layout_object);
```

Reply:
Realized that I should be using `ResourceReferenceChanged` before seeing this comment 😅